src/mesido/workflows/io/write_output.py

```diff
@@ -321,11 +321,16 @@ def _name_to_asset(name):
         kpis_top_level = esdl.KPIs(id=str(uuid.uuid4()))
         heat_source_energy_wh = {}
         asset_capex_breakdown = {}
-        asset_opex_breakdown = {}
+        asset_opex_breakdown = {}  # yearly cost
         tot_install_cost_euro = 0.0
         tot_invest_cost_euro = 0.0
-        tot_variable_opex_cost_euro = 0.0
-        tot_fixed_opex_cost_euro = 0.0
+        tot_variable_opex_cost_euro = 0.0  # yearly cost
+        tot_fixed_opex_cost_euro = 0.0  # yearly cost
+
+        # over entire lifetime costs
+        asset_lifetime_opex_breakdown = {}
+        tot_lifetime_variable_opex_cost_euro = 0.0
```

Comment:
assets with different lifetimes, will result in different total opex, while actually for the horizon we are optimising  (30 years), you would then also replace assets which only have a 10 year life time, 3 times.
The total will therefore not necessarily say something about the opex for the entire system for those 30 years.
If that is what you're intending to say, then we should use the horizon instead of the asset technicallifetime.

Reply:
I agree this was overlooked. Thanks. I have now changed it to total time horizon (over which TCO is minimized) instead of lifetime. So now the opex values for all assets (even if they have different lifetime values themselves) will be calculated over the fixed 30 year (grow workflow->parameters["number_of_years"]->self._number_of_years)